code/Weapons/Gadget/Components/AnvilPhysicsGadgetComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace Grubs;
+
+[Prefab]
+public partial class AnvilPhysicsGadgetComponent : GadgetComponent
+{
+	[Prefab, Net]
+	public bool LockXAxis { get; set; } = true;
+
+	[Prefab, Net]
+	public bool AffectedByWind { get; set; } = false;
+
+	[Prefab, ResourceType( "sound" )]
+	public string CollisionSound { get; set; }
+
+	[Prefab, Net, Predicted]
+	public int Bounces { get; set; } = 0;
+
+	[Prefab, Net]
+	public float BounceForce { get; set; } = 100.0f;
+
+	private bool _isGrounded;
+	private bool _wasGrounded;
+
+	public override bool IsResolved()
+	{
+		return Bounces == 0 && Gadget.Velocity.IsNearlyZero( 2.5f );
+	}
+
+	public override void OnUse( Weapon weapon, int charge )
+	{
+		if ( GrubsConfig.WorldTerrainType is GrubsConfig.TerrainType.Texture )
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsGame.Instance.Terrain.WorldTextureHeight + 192f );
+		else
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsConfig.TerrainHeight + 192f );
+	}
+
+	public override void Simulate( IClient client )
+	{
+		// Apply gravity.
+		Gadget.Velocity -= new Vector3( 0, 0, 400 ) * Time.Delta;
+
+		var helper = new MoveHelper( Gadget.Position, Gadget.Velocity );
+		helper.Trace = helper.Trace
+			.Size( Gadget.CollisionBounds )
+			.WithAnyTags( Tag.Player, Tag.Solid )
+			.WithoutTags( Tag.Dead )
+			.UseHitboxes( true );
+
+		helper.TryMove( Time.Delta );
+		Gadget.Position = helper.Position;
+		Gadget.Velocity = helper.Velocity;
+		if ( LockXAxis )
+			Gadget.Velocity = Gadget.Velocity.WithX( 0 );
+
+		var trCollision = Trace.Ray( Gadget.Position, Gadget.Position )
+				.Size( Gadget.CollisionBounds * 1.1f ) // Slightly increase to make sure it collides.
+				.Ignore( Gadget )
+				.WithAnyTags( Tag.Player, Tag.Solid )
+				.WithoutTags( Tag.Shard )
+				.UseHitboxes( true )
+				.Run();
```

Comment:
Are we able to use the move helper trace down in this case?

Reply:
Originally, this was using tracing down with MoveHelper.TraceDirection(), however this was swapped to match ExplosiveGadgetComponent's collision check. This is because when I swapped to ExplosiveGadgetComponent checking collision/triggering an explosion, the explosion collision detection was bigger and therefore would always trigger before AnvilPhysics ever detected a collision.

Essentially, both collision checks must be the same, so off the top of my head we can either swap the collision check back just for AnvilPhysics and call Explode manually, or change the collision check on both AnvilPhysics AND ExplosiveGadget to be the same MoveHelper.TraceDirection( Vector3.Down ).

Reply:
I think I'd prefer `change the collision check on both AnvilPhysics AND ExplosiveGadget to be the same MoveHelper.TraceDirection( Vector3.Down ).`

What do you think @trundlr

Reply:
Does not matter to me, gut reaction says use the same check on both.